app/process/handler/consensus-message/consensus-message.go

```diff
@@ -89,21 +111,95 @@ func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
 	}
 
 	err = cmh.repository.Create(&message.TransactionMessage{
-		TransactionId: m.TransactionId,
-		EthAddress:    m.EthAddress,
-		Amount:        m.Amount,
-		Fee:           m.Fee,
-		Signature:     m.Signature,
-		Hash:          hexHash,
+		TransactionId:        m.TransactionId,
+		EthAddress:           m.EthAddress,
+		Amount:               m.Amount,
+		Fee:                  m.Fee,
+		Signature:            m.Signature,
+		Hash:                 hexHash,
+		SignerAddress:        address.String(),
+		TransactionTimestamp: m.TransactionTimestamp,
 	})
 	if err != nil {
 		return errors.New(fmt.Sprintf("Could not add Transaction Message with Transaction Id and Signature - [%s]-[%s]", m.TransactionId, m.Signature))
 	}
 
-	log.Printf("Successfully verified and persisted TX with ID [%s]\n", m.TransactionId)
+	log.Infof("Successfully verified and persisted TX with ID [%s]", m.TransactionId)
+
+	txSignatures, err := cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	requiredSigCount := len(cmh.validAddresses)/2 + len(cmh.validAddresses)%2
+	log.Infof("Required signatures: [%v]", requiredSigCount)
+
+	txSignatures, err = cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	if len(txSignatures) < requiredSigCount {
+		log.Warnf(fmt.Sprintf("Insignificant amount of Transaction Signatures for Transaction [%s] - [%d] signatures", m.TransactionId, len(txSignatures)))
```

Comment:
If the collected signatures are not enough **yet**, I think that we should log info not warning. It is normal to have a period in which signatures are not the required amount